From patchwork Fri Oct 22 17:24:03 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Mark Yacoub <markyacoub@chromium.org>
X-Patchwork-Id: 12578269
Return-Path: 
 <SRS0=WF8g=PK=lists.infradead.org=linux-mediatek-bounces+linux-mediatek=archiver.kernel.org@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id AD5BEC433EF
	for <linux-mediatek@archiver.kernel.org>;
 Fri, 22 Oct 2021 17:24:44 +0000 (UTC)
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id 54EB860F4F
	for <linux-mediatek@archiver.kernel.org>;
 Fri, 22 Oct 2021 17:24:44 +0000 (UTC)
DMARC-Filter: OpenDMARC Filter v1.4.1 mail.kernel.org 54EB860F4F
Authentication-Results: mail.kernel.org;
 dmarc=fail (p=none dis=none) header.from=chromium.org
Authentication-Results: mail.kernel.org;
 spf=none smtp.mailfrom=lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:Cc
	:To:From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:
	List-Owner; bh=nVZz6DescsSD2DK0WMHjNfr+kMVdZKlgEbVZ0UTJGMY=; b=ylPJs2CtwV115G
	DBg8Fgu1Ju7Otn4OkuY/w9hAuq4xaJ3AE0RtSn7vW8SWnoxFJuaoZmcYk5bwsAUnrf48H+uZkBpcw
	5fZTgjq2mRUMHpmHN+8ZnJSgWf2myNv6PnMFaza1u5I1QzKQZtNW4U8AY89qp2w3/tYPBar0S1sH/
	AdebI3gUo/lw+7yRI0F9fEEvOBoWcvepqLPsE82S4Eyaf1FBKcxVOsmTOvLim+TA3PLRJdAF8IXUF
	mwvSVIapJg3XNU5WJ3DHiCxOwctS2eg1+IXBCjp9+BvLHfbZbXYZy8dwGLs6qNtII2c3x6ZpMy1KW
	LDBD7t78DyRI0BPgq4BA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1mdyHS-00Bcl1-5L; Fri, 22 Oct 2021 17:24:34 +0000
Received: from mail-qk1-x735.google.com ([2607:f8b0:4864:20::735])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1mdyHF-00BciK-6f
 for linux-mediatek@lists.infradead.org; Fri, 22 Oct 2021 17:24:23 +0000
Received: by mail-qk1-x735.google.com with SMTP id d15so5193681qkj.10
 for <linux-mediatek@lists.infradead.org>;
 Fri, 22 Oct 2021 10:24:20 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=chromium.org;
 s=google;
 h=from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=bhRqLM6yW3BElMhr6p8JxBlLhkjCyrvm/fO8BA6PxAc=;
 b=n0REBVugNN6h/yPp0kthsRhifZEkvICP1H0DNb4lHJcDlVhDv7nMaKJ2Vhs+3mG8wb
 P+9hayC3ZcaTRQbQ6Yg/emEujycTI0bGB0+E/8Au1iQ2EaTxB9N8pARvjQsu5l7fjzyx
 LXthrCGguKWkBeGh5u6EqjmXuYKmprV7mEMPQ=
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20210112;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=bhRqLM6yW3BElMhr6p8JxBlLhkjCyrvm/fO8BA6PxAc=;
 b=tl5JGQkcdttJwVvC0Za/U/MlM5pVKoK3gS8RaDTbnRD+Pnwks8pyeYGkkTDZv8LJHU
 hsE51AdUccszQSCjyS7w1Ux7tRvBVh1d0/JEuvEjF/wanTr+/jBJ6QSieoKgioZK0a7M
 yQcjYo0iRJFdx82LQFFTNPnC88VX/QgFBOpAXbRe8JhqHQJZ9TZZ4/1lyHofqImbSTpr
 IFxcsk014onATivERTYO1kc9Yx0Yr7GJJqwHH4li837JroG+sbZ5BCoxMsLaV4u5KqSs
 xC0SQi2KlgHuxRiltxxxLyaoZoKrzJR/sZUFqOw/qzYz/C/CBYG5B4FASTouV138r9m8
 5qHw==
X-Gm-Message-State: AOAM533+/ZCyvF2qSFjDto8DDKDYK/yr6t3mgvEG8uTl2A2400g6pQ9e
 S/eg788OpkSm3hDidj1RXMx1Z4CkPztWgWxbJFE=
X-Google-Smtp-Source: 
 ABdhPJw5+Z3gbQpB89P9MYXrYchvZhhA7n6NVaZf4RXniPOT8qvw6+uopCRuEw1xS7/oSzdg0o1iFw==
X-Received: by 2002:a37:a2c1:: with SMTP id l184mr1122332qke.71.1634923459566;
 Fri, 22 Oct 2021 10:24:19 -0700 (PDT)
Received: from markyacoub.nyc.corp.google.com
 ([2620:0:1003:314:6c36:8d9f:c50a:b0f0])
 by smtp.gmail.com with ESMTPSA id m17sm4607611qtx.62.2021.10.22.10.24.18
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Fri, 22 Oct 2021 10:24:18 -0700 (PDT)
From: Mark Yacoub <markyacoub@chromium.org>
To: linux-mediatek@lists.infradead.org
Cc: seanpaul@chromium.org, Mark Yacoub <markyacoub@google.com>,
 Mark Yacoub <markyacoub@chromium.org>,
 Chun-Kuang Hu <chunkuang.hu@kernel.org>,
 Philipp Zabel <p.zabel@pengutronix.de>, David Airlie <airlied@linux.ie>,
 Daniel Vetter <daniel@ffwll.ch>, Matthias Brugger <matthias.bgg@gmail.com>,
 dri-devel@lists.freedesktop.org, linux-arm-kernel@lists.infradead.org,
 linux-kernel@vger.kernel.org
Subject: [PATCH] mediatek: Set default value for Panel Orientation connector
 prop.
Date: Fri, 22 Oct 2021 13:24:03 -0400
Message-Id: <20211022172413.195559-1-markyacoub@chromium.org>
X-Mailer: git-send-email 2.33.0.1079.g6e70778dc9-goog
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20211022_102421_262553_15ED4049 
X-CRM114-Status: UNSURE (   9.74  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-mediatek@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: <linux-mediatek.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-mediatek>,
 <mailto:linux-mediatek-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-mediatek/>
List-Post: <mailto:linux-mediatek@lists.infradead.org>
List-Help: <mailto:linux-mediatek-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-mediatek>,
 <mailto:linux-mediatek-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-mediatek" <linux-mediatek-bounces@lists.infradead.org>
Errors-To: 
 linux-mediatek-bounces+linux-mediatek=archiver.kernel.org@lists.infradead.org

From: Mark Yacoub <markyacoub@google.com>

[Why]
Creating the prop uses UNKNOWN as the initial value, which is not a
supported value if the props is to be supported.

[How]
Set the panel orientation default value to NORMAL right after creating
the prop.

Tested on Jacuzzi(MTK)
Fixes IGT@kms_properties@get_properties-sanity-{atomic,non-atomic}

Signed-off-by: Mark Yacoub <markyacoub@chromium.org>
---
 drivers/gpu/drm/mediatek/mtk_dsi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/mediatek/mtk_dsi.c b/drivers/gpu/drm/mediatek/mtk_dsi.c
index 0ad7157660afa..b7c2528a8f41c 100644
--- a/drivers/gpu/drm/mediatek/mtk_dsi.c
+++ b/drivers/gpu/drm/mediatek/mtk_dsi.c
@@ -1039,6 +1039,8 @@ static int mtk_dsi_encoder_init(struct drm_device *drm, struct mtk_dsi *dsi)
 		DRM_ERROR("Unable to init panel orientation\n");
 		goto err_cleanup_encoder;
 	}
+	drm_connector_set_panel_orientation(dsi->connector,
+					    DRM_MODE_PANEL_ORIENTATION_NORMAL);
 
 	drm_connector_attach_encoder(dsi->connector, &dsi->encoder);
 
